packages/common.vm/tools/vm.common/vm.common.psm1

```diff
@@ -886,3 +886,379 @@ RAW_TOOLS_DIR: ${Env:RAW_TOOLS_DIR}
     VM-Write-Log "INFO" "$($survey -f $osInfo, $memInfo, $diskInfo, $avInfoFormatted, $psInfo, $psInfoClr, $chocoInfo, $boxstarerInfo, $installedPackages, $envVars)"
 }
 
+function VM-Remove-Appx-Package {
+# Function for removing Apps
+    param(
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$appName
+    )
+
+    try {
+        # Check if the app is installed
+        $installedPackage = Get-AppxPackage -Name $appName -ErrorAction SilentlyContinue
+        
+        if ($installedPackage) {
+            $packageFullName = $installedPackage.PackageFullName
+            $result = Remove-AppxPackage -Package $packageFullName -ErrorAction Stop
+
+            if ($null -eq $result) {
+                VM-Write-Log "INFO" "[+] Installed $appName has been successfully removed."
+            } else {
+                VM-Write-Log "ERROR" "[+] Failed to remove installed app $appName."
+            }
+        }
+        else {
+            VM-Write-Log "WARN" "[+] Installed $appName not found on the system."
+        }
+        # Check if the app is provisioned
+        $provisionedPackage = Get-AppxProvisionedPackage -Online | Where-Object { $_.DisplayName -eq $appName } -ErrorAction SilentlyContinue
+        if ($provisionedPackage) {
+            $result = Remove-AppxProvisionedPackage -PackageName $provisionedPackage.PackageName -Online  
```

Comment:
Similarly as before, you can't use the result of`Remove-AppxProvisionedPackage` to text if it succeeded. An example:
 
![Screenshot 2023-09-06 at 13 09 45](https://github.com/mandiant/VM-Packages/assets/16052290/d391fc0b-d40b-45a1-8e66-13d53f8ec399)